heat/core/arithmetics.py

```diff
@@ -63,6 +65,78 @@ def add(t1, t2):
     return operations.__binary_op(torch.add, t1, t2)
 
 
+def diff(a, n=1, axis=-1):
+    """
+    Calculate the n-th discrete difference along the given axis.
+    The first difference is given by out[i] = a[i+1] - a[i] along the given axis, higher differences are calculated by using diff recursively.
```

Comment:
Shouldn't that be a[i + n] instead of a[i + 1]?

Reply:
i thought so at first too. but numpy diff is just with the next element. the n is essentially just a recursion and not a difference in how far to go before taking the subtraction